dss/util/aws/__init__.py

```diff
@@ -0,0 +1,32 @@
+import json
+import botocore
+from . import resources, clients
+
+class ARN:
+    fields = "arn partition service region account_id resource".split()
+    _default_region, _default_account_id, _default_iam_username = None, None, None
+    def __init__(self, arn="arn:aws::::", **kwargs):
+        self.__dict__.update(dict(zip(self.fields, arn.split(":", 5)), **kwargs))
+        if "region" not in kwargs and not self.region:
+            self.region = self.get_region()
+        if "account_id" not in kwargs and not self.account_id:
+            self.account_id = self.get_account_id()
+
+    @classmethod
+    def get_region(cls):
+        if cls._default_region is None:
+            cls._default_region = botocore.session.Session().get_config_variable("region")
```

Comment:
I'm pretty sure this is always set in a functioning environment, because if it's not, botocore will throw an error on any AWS operation. Can you recall the details of when it was not set?